pkg/watch/replay.go

```diff
@@ -121,94 +121,96 @@ func (r *replayTracker) replayWaitCh(gvk schema.GroupVersionKind) chan struct{}
 // replayEventsLoop processes requests to start and stop replaying events for
 // registrars that join an existing informer and need historical data.
 // Events for each registrar will be listed and replayed in an independent goroutine.
-func (wm *Manager) replayEventsLoop() error {
-	var wg sync.WaitGroup
-	defer wg.Wait()
+func (wm *Manager) replayEventsLoop(ctx context.Context) func() error {
+	return func() error {
+		var wg sync.WaitGroup
+		defer wg.Wait()
 
-	ctx, cancel := syncutil.ContextForChannel(wm.stopped)
-	defer cancel()
+		ctx, cancel := syncutil.ContextForChannel(ctx, wm.stopped)
```

Comment:
I don't think we need to merge these two channels - `ctx` should be enough as `Start` only signals `wm.stopped` after `ctx` is canceled. Similarly - `replayEvents` doesn't need to consider `wm.stopped` anymore. There may be more instances of this - I'd rather have a simpler cancelation tree if possible.

Reply:
Ah, I see what you mean. Done.